src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -211,7 +227,46 @@ class CanvasImageLayerRenderer extends CanvasLayerRenderer {
       }
     }
 
-    const img = this.getImage();
+    /**
+     * @type {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement|ImageBitmap}
+     */
+    let img = image.getImage();
+    if (img !== this.previousImg_ && this.imageBitmap_) {
+      this.imageBitmap_.close();
+      this.imageBitmap_ = null;
+    }
+    if (
+      img === this.previousImg_ &&
+      !this.getLayer().getSource()?.scaleable &&
+      !(img instanceof HTMLCanvasElement) &&
+      img.width * img.height < 268435456
+    ) {
+      if (!this.imageBitmap_) {
+        // First draw image to a canvas to convert the image data to the fastest format possible
+        // for repeatedly calling drawImage(), then store the result in an image bitmap to avoid
+        // canvas memory shortage in Safari.
+        const context = this.conversionContext_;
+        context.canvas.width = img.width;
+        context.canvas.height = img.height;
+        context.drawImage(img, 0, 0);
+        createImageBitmap(context.canvas)
+          .then((imageBitmap) => {
+            this.imageBitmap_ = imageBitmap;
+          })
+          .catch(() => {
+            // cannot allocate image bitmap
+          })
+          .finally(() => {
+            context.canvas.width = 1;
+            context.canvas.height = 1;
+          });
```

Comment:
At this point, it looks like `this.imageBitmap_` would still be falsey and the function will proceed with rendering whatever `img` is.  Then on the next call to `renderFrame`, we would find ourselves here again, having called `createImageBitMap` again.  Does this matter?

Reply:
In my tests it did not matter, because `this.imageBitmap_` would be an `ImageBitmap` already by the time `renderFrame` is called the next time. Anyway, I added a commit to change the behavior so that the temporary canvas is used until the `ImageBitmap` is available.